nym-vpn-x/src-tauri/src/grpc/client.rs

```diff
@@ -224,6 +224,53 @@ impl GrpcClient {
         Ok(())
     }
 
+    /// Watch VPN status updates
+    #[instrument(skip_all)]
+    pub async fn watch_vpn_status(&self) -> Result<()> {
+        let mut vpnd = self.vpnd().await?;
+
+        let request = Request::new(Empty {});
+        let mut stream = vpnd
+            .listen_to_connection_status(request)
+            .await
+            .inspect_err(|e| {
+                error!("listen_to_connection_status failed: {}", e);
+            })?
+            .into_inner();
+
+        let (tx, mut rx) = mpsc::channel(32);
+        tokio::spawn(async move {
```

Comment:
It looks like you are anyway running an event loop below, so I'm not sure you need to spawn this task. Can't you just look over stream directly below where you are currently looking over rx?

Reply:
I spawn the task to not block current thread during watching the stream, as it needs to be iterated in a loop. Also, I need to use a channel because I'll need to use (borrow/move) the `AppHandle` once we'll handle status update (same logic as in `watch_vpn_state` function above).
So unless I missed something, no other choice than having to use a channel.

Reply:
But it looks like the function is blocking on the while loop?

Reply:
Just try deleting the whole tokio::spawn thingie and replace 
```
while let Some(status) = rx.recv().await {
```
with
```
while let Ok(Some(status)) = stream.message().await {
```